rcp/org.testeditor.rcp4.uatests/src/org/testeditor/fixture/swt/SWTFixture.java

```diff
@@ -0,0 +1,114 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.fixture.swt;
+
+import static org.eclipse.swtbot.swt.finder.matchers.WidgetMatcherFactory.widgetOfType;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swt.widgets.Widget;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.testeditor.fixture.core.interaction.FixtureMethod;
+
+public class SWTFixture {
+
+	Logger logger = Logger.getLogger(SWTFixture.class);
+	SWTWorkbenchBot bot = new SWTWorkbenchBot();
+
+	/**
+	 * Method for debugging the AUT. A first implementation of an widget
+	 * scanner. all widget s on current SWT-App will be printed with their id's
+	 * in the swtbot.log with the trace-level. This Method is in work and can be
+	 * extended and modified step by step.
+	 * 
+	 */
+	@FixtureMethod
+	public void reportWidgets() {
+		logger.trace("analyzeWidgets start");
+		logger.trace("---------------------------------------------");
+
+		Display.getDefault().syncExec(new Runnable() {
+			@Override
+			public void run() {
+
+				try {
+					List<? extends Widget> widgets = bot.widgets(widgetOfType(Widget.class));
+
+					StringBuilder sb = new StringBuilder();
+
+					for (Widget widget : widgets) {
+
+						if (widget instanceof Table) {
+							sb.append("\n >>> Table gefunden mit " + ((Table) widget).getItems().length + " Zeilen !");
+						}
+
+						sb.append("widgetId: " + widget.getData("org.eclipse.swtbot.widget.key"));
+						sb.append(" widgetClass: " + widget.getClass().getSimpleName());
+						try {
+							Method[] methods = widget.getClass().getMethods();
+							boolean foundGetText = false;
+							for (Method method : methods) {
+								if (method.getName().equals("getText")) {
+									foundGetText = true;
+								}
+							}
+							if (foundGetText) {
+
+								Method method = widget.getClass().getMethod("getText", new Class[] {});
+								sb.append("\n >>> text value: " + method.invoke(widget, new Object[] {}));
+							}
+						} catch (Exception e) {
+							logger.error(">>>>>>> no text", e);
+						}
+						sb.append(" widget: " + widget).append("\n");
+					}
+
+					logger.trace(sb.toString());
+
+				} catch (Exception e) {
+					logger.error("ERROR " + e.getMessage());
+				}
+
+			}
+		});
+		logger.trace("analyzeWidgets end");
+		logger.trace("---------------------------------------------");
+
+	}
+
+	@FixtureMethod
+	public void wait(String seconds) throws NumberFormatException, InterruptedException {
+		logger.trace("wait for: " + seconds);
```

Comment:
This line should read: `logger.trace("Wait for: {}", seconds)` so that the concatenation is not executed when trace is not enabled. Apart from that, I think the log level should be info.